util/internal/test/common/src/main/java/org/hibernate/search/util/impl/test/extension/parameterized/ParameterizedClassTestMethodDiscoverer.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.util.impl.test.extension.parameterized;
+
+import static org.hibernate.search.util.impl.test.extension.parameterized.ParameterizedClassUtils.findParameters;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.platform.commons.util.AnnotationUtils;
+
+class ParameterizedClassTestMethodDiscoverer {
+
+	private ParameterizedClassTestMethodDiscoverer() {
+	}
+
+	public static List<ParameterizedTestMethodInvoker> discover(Class<?> testClass, ExtensionContext extensionContext) {
+		List<ParameterizedTestMethodInvoker> testMethods = new ArrayList<>();
+		for ( Method method : testClass.getDeclaredMethods() ) {
+			if ( ParameterizedClassUtils.isActualTestMethodToExecute( method ) ) {
+				testMethods.addAll( toTestMethods( method, extensionContext ) );
+			}
```

Comment:
I think there's room for some checks to avoid dumb mistakes here (I guarantee I'll make one someday).

If a method in a `@ParameterizedClass` test class is annotated with any JUnit annotation beside those recognized by `ParameterizedClassExtension` (i.e. `@Test`, `@ParameterizedTest`, `@ParameterizedSetup`, and the various `@MethodSource`/`@CsvSource`/etc.) we should probably throw an exception because nobody will handle that annotation.

Also I think `@Nested` classes probably won't work, so we should throw an exception if there is any?